applications/GeoMechanicsApplication/custom_elements/transient_Pw_line_element.h

```diff
@@ -354,31 +293,110 @@ class KRATOS_API(GEO_MECHANICS_APPLICATION) TransientPwLineElement : public Elem
         array_1d<double, TNumNodes * TDim> volume_acceleration;
         GeoElementUtilities::GetNodalVariableVector<TDim, TNumNodes>(
             volume_acceleration, GetGeometry(), VOLUME_ACCELERATION);
-        array_1d<double, TDim> body_acceleration;
 
         array_1d<double, TNumNodes> fluid_body_vector = ZeroVector(TNumNodes);
         for (unsigned int integration_point_index = 0;
              integration_point_index < GetGeometry().IntegrationPointsNumber(GetIntegrationMethod());
              ++integration_point_index) {
+            array_1d<double, TDim> body_acceleration;
             GeoElementUtilities::InterpolateVariableWithComponents<TDim, TNumNodes>(
-                body_acceleration, rNContainer, volume_acceleration, integration_point_index);
+                body_acceleration, N_container, volume_acceleration, integration_point_index);
 
             array_1d<double, TDim> tangent_vector = column(J_container[integration_point_index], 0);
             tangent_vector /= norm_2(tangent_vector);
 
             array_1d<double, 1> projected_gravity = ZeroVector(1);
             projected_gravity(0) = MathUtils<double>::Dot(tangent_vector, body_acceleration);
-            const auto N         = Vector{row(rNContainer, integration_point_index)};
+            const auto N         = Vector{row(N_container, integration_point_index)};
             double     RelativePermeability =
                 mRetentionLawVector[integration_point_index]->CalculateRelativePermeability(RetentionParameters);
             fluid_body_vector +=
                 r_properties[DENSITY_WATER] * RelativePermeability *
-                prod(prod(rShapeFunctionGradients[integration_point_index], constitutive_matrix), projected_gravity) *
-                rIntegrationCoefficients[integration_point_index] / r_properties[DYNAMIC_VISCOSITY];
+                prod(prod(shape_function_gradients[integration_point_index], constitutive_matrix), projected_gravity) *
+                integration_coefficients[integration_point_index] / r_properties[DYNAMIC_VISCOSITY];
         }
         return fluid_body_vector;
     }
 
+    std::unique_ptr<Calculator> CreateCalculator(const CalculationContribution& rContribution,
+                                                 const ProcessInfo&             rCurrentProcessInfo)
+    {
+        switch (rContribution) {
+        case CalculationContribution::Permeability:
+            return std::make_unique<PermeabilityCalculator>(CreatePermeabilityInputProvider());
+        case CalculationContribution::Compressibility:
+            return std::make_unique<CompressibilityCalculator>(CreateCompressibilityInputProvider(rCurrentProcessInfo));
+        default:
+            KRATOS_ERROR << "Unknown contribution" << std::endl;
```

Comment:
Would it be useful to extent Check() with checking the input with calculators' names?

Reply:
I think `Check` shall check all inputs; however, it will create a duplication if the calculators are created in `CalculateLocalSystem`.